### PR TITLE
Hotfix/mail listener attachments

### PR DESCRIPTION
**Pull request by conitrade-as**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] Ready

## Related Issues
- N/A

## Description
This change set fixes the Mail Listener v2 to work with attachments if the commands are invoked from the war room or from within playbooks. The existing implementation did not output attachments which made them non-existing for e.g. downstream playbook steps.

Also, specifying e-mail identifiers as integers is implemented and allows the commands to actually work.

### CODE REVIEW

**Comment by rshunim** [comment]
Hi @conitrade-as, thanks for your contribution!
I see only one file is changed.
1. When editing an integration, you should add Release Note file and update to a new version in the pack_metadata file.
2. Instead of converting an argument with int(), we have a function called `arg_to_number()`
3. Please fix all the errors in the build (make it green, passes)

Then I'll review it again

Thanks a lot!

**Review by rshunim — COMMENTED** [review]
See my comments

**Comment by conitrade-as** [comment]
Fixed issues and rebased changes.

File: Packs/MailListener/ReleaseNotes/1_0_58.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Mail Listener v2
+
+- Output attachments to warroom
```
Comment (rshunim) [inline Packs/MailListener/ReleaseNotes/1_0_58.md:6]:
```suggestion
- Added support for output attachments to war room.
```

File: Packs/MailListener/ReleaseNotes/1_0_58.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- Convert message ids to ints
```
Comment (rshunim) [inline Packs/MailListener/ReleaseNotes/1_0_58.md:7]:
```suggestion
- Improved implementation by converting message ids to ints.
```

**Review by rshunim — COMMENTED** [review]
Good job!
just a last thing, please add a UT which checking the new logic when `output_to_warroom` is True.

Then let's set a demo together.
You can ping me via Slack too (Roei Shunim)

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 3]
```diff
@@ -639,10 +641,10 @@ def main():     # pragma: no cover
                                            _limit=limit))
             elif demisto.command() == 'mail-listener-get-email':
                 return_results(get_email(client=client,
-                                         message_id=args.get('message-id')))
+                                         message_id=arg_to_number(args.get('message-id', 0))))
```
Comment (rshunim) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:644]:
```suggestion
                                         message_id=arg_to_number(args.get('message-id')) or 0))
```

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 4]
```diff
@@ -639,10 +641,10 @@ def main():     # pragma: no cover
...
             elif demisto.command() == 'mail-listener-get-email-as-eml':
                 return_results(get_email_as_eml(client=client,
-                                                message_id=args.get('message-id')))
+                                                message_id=arg_to_number(args.get('message-id', 0))))
```
Comment (rshunim) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:647]:
```suggestion
                                                message_id=arg_to_number(args.get('message-id')) or 0))
```

Comment (rshunim) [reply Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:647]: for avoiding mypy error in the pre-commit

**Comment by rshunim** [comment]
@ShirleyDenkberg Please review it

File: Packs/MailListener/ReleaseNotes/1_0_58.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### Mail Listener v2
+
+- Added support for output attachments to war room.
```
Comment (ShirleyDenkberg) [inline Packs/MailListener/ReleaseNotes/1_0_58.md:6]:
```suggestion
- Added support for output attachments to the War Room.
```

File: Packs/MailListener/ReleaseNotes/1_0_58.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+- Improved implementation by converting message ids to ints.
```
Comment (ShirleyDenkberg) [inline Packs/MailListener/ReleaseNotes/1_0_58.md:7]:
```suggestion
- Improved implementation by converting message IDs to ints.
```

**Comment by ShirleyDenkberg** [comment]
@rshunim @samuelFain @YuvHayun @altmannyarden @idovandijk Doc review completed.

**Review by rshunim — APPROVED** [review]
Good Job!

**Pull request merged**